src/test/java/uk/ac/ebi/eva/pipeline/io/readers/VcfReaderTest.java

```diff
@@ -0,0 +1,205 @@
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSource;
+import org.opencb.biodata.models.variant.VariantSourceEntry;
+import org.opencb.biodata.models.variant.VariantStudy;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.file.FlatFileParseException;
+import org.springframework.batch.test.MetaDataInstanceFactory;
+
+import uk.ac.ebi.eva.test.utils.JobTestUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * {@link VcfReader}
+ * input: a Vcf file
+ * output: a list of variants each time its `.read()` is called
+ */
+public class VcfReaderTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void shouldReadAllLines() throws Exception {
+        ExecutionContext executionContext = MetaDataInstanceFactory.createStepExecution().getExecutionContext();
+
+        // input vcf
+        final String inputFilePath = "/small20.vcf.gz";
+        String input = VcfReaderTest.class.getResource(inputFilePath).getFile();
+
+        String fileId = "5";
+        String studyId = "7";
+        String studyName = "study name";
+        VariantSource source = new VariantSource(input, fileId, studyId, studyName, VariantStudy.StudyType.COLLECTION,
+                                                 VariantSource.Aggregation.NONE);
+
+        VcfReader vcfReader = new VcfReader(source, input);
+        vcfReader.setSaveState(false);
+        vcfReader.open(executionContext);
+
+        consumeReader(input, source, vcfReader);
+    }
+
+    @Test
+    public void invalidFileShouldFail() throws Exception {
+        ExecutionContext executionContext = MetaDataInstanceFactory.createStepExecution().getExecutionContext();
+
+        // input vcf
+        final String inputFilePath = "/wrong_no_alt.vcf.gz";
+        String inputFile = VcfReaderTest.class.getResource(inputFilePath).getFile();
+
+        String fileId = "5";
+        String studyId = "7";
+        String studyName = "study name";
+        VariantSource source = new VariantSource(inputFile, fileId, studyId, studyName,
+                                                 VariantStudy.StudyType.COLLECTION, VariantSource.Aggregation.NONE);
+
+        VcfReader vcfReader = new VcfReader(source, inputFile);
+        vcfReader.setSaveState(false);
+        vcfReader.open(executionContext);
+
+        // consume the reader and check that a wrong variant raise an exception
+        exception.expect(FlatFileParseException.class);
+        while (vcfReader.read() != null) {
+        }
```

Comment:
I would let this as is to enforce the rule of putting always the braces in `while`, `if` and `else` blocks